claim_batch/services.py

```diff
@@ -546,91 +564,107 @@ def do_process_batch(audit_user_id, location_id, period, year):
+
+    if period_rate > 0 or period_rate_ip > 0 or period_rate_op > 0:
         for item in items:
-            item_quatity = item.qty_approved ? item.qty_approved : item.qty_provided # if qty_approved not null
-            item_price = item.price_approved ? item.price_approved : (item.price_adjusted ? item.price_adjusted : item.price_asked)  # if price_approved not null
-            if is_hospital_claim(work_data.product, item.claim):
+            item_quatity = item.qty_approved if item.qty_approved else item.qty_provided
+            item_price = item.price_approved if item.price_approved else (item.price_adjusted if item.price_adjusted else item.price_asked)  # if price_approved not null
+            if is_hospital_claim(product, item.claim):
                 value_hospital = item_price * item_quatity
             else:
                 value_non_hospital = item_price * item_quatity
 
-
         for service in services:
-            service_quatity = service.qty_approved ? service.qty_approved : service.qty_provided # if qty_approved not null
-            service_price = service.price_approved ? service.price_approved : (service.price_adjusted ? service.price_adjusted : service.price_asked)  # if price_approved not null
-            if is_hospital_claim(work_data.product, service.claim):
+            service_quatity = service.qty_approved if service.qty_approved else service.qty_provided
+            service_price = service.price_approved if service.price_approved else (service.price_adjusted if service.price_adjusted else service.price_asked)  # if price_approved not null
+            if is_hospital_claim(product, service.claim):
                 value_hospital = service_price * service_quatity
             else:
-                value_non_hospital = item_price * item_quatity
+                value_non_hospital = service_price * service_quatity
             # calculate the index based on product config
 
         # creeeate i/o index OR in and out patien index
-        if period_rate>0:
-            index = (value_non_hospital + value_hospital) > 0 ? allocated_contributions/(value_non_hospital + value_hospital) : 1
-            create_index(work_data.product, index,'B',periode_type, period_id)
-        else:
-            elif period_rate_ip>0:
-                index_ip = value_hospital >0 ? allocated_contributions / value_hospital :1
-                create_index(work_data.product, index_ip ,'I',periode_type, period_id)
-            elif period_rate_op>0:
-                index_op = value_non_hospital >0 ? allocated_contributions / value_non_hospital :1
-                create_index(work_data.product,  index_op ,'O',periode_type, period_id)
+        if period_rate > 0:
+            index = allocated_contributions/value_non_hospital if value_non_hospital + value_hospital > 0 else 1
+            create_index(product, index,'B',period_type, period_id)
+        elif period_rate_ip > 0:
```

Comment:
it needs to be 2 ifs because it is either general or IP + OP

Reply:
ok

Reply:
@delcroip And what about this  if "elif period_rate_op > 0:"?

Reply:
ok, I added two ifs according to the pseudocode. It was my mistake. Now should be ok